internal/controller/progressive/progressive.go

```diff
@@ -46,6 +46,12 @@ type progressiveController interface {
 
 	// AssessUpgradingChild determines if upgrading child is determined to be healthy, unhealthy, or unknown
 	AssessUpgradingChild(ctx context.Context, existingUpgradingChildDef *unstructured.Unstructured) (apiv1.AssessmentResult, error)
+
+	// ScaleDownPromotedChildSourceVertices scales down by half the promoted child source vertices pods
+	ScaleDownPromotedChildSourceVertices(ctx context.Context, rolloutObject ctlrcommon.RolloutObject, promotedChildDef *unstructured.Unstructured, c client.Client) (map[string]apiv1.ScaleValues, bool, error)
+
+	// ScalePromotedChildSourceVerticesToDesiredValues scales back up to the desired scale values the promoted child source vertices pods
+	ScalePromotedChildSourceVerticesToDesiredValues(ctx context.Context, rolloutObject ctlrcommon.RolloutObject, promotedChildDef *unstructured.Unstructured, c client.Client) error
```

Comment:
We could probably include this in the first PR, but I'd love to not have these in the `progressiveController` interface, since it's not a hard requirement that scaling be part of the Rollouts, or that the progressiveController even have "vertices". I'll think of different options for how we might do that.

Reply:
One option:
Change these to be more generic:
`PreparePromotedChildForUpgrading()` and 
`PreparePromotedChildForUpgradingCompletion()`

We can still have your `scaleDownPromotedChild()` and `scaleUpPromotedChild()` functions, but just have them called from these interface functions for Pipeline and MV.